src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -1134,7 +1140,234 @@ def test_when_zero_pageSize_then_InputValidationException_is_raised(
+            an_entity_trait_set,
+            page_size,
+            access.RelationsAccess.kRead,
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert isinstance(actual_pager, EntityReferencePager)
+        actual_pager.get()  # Assert that pager is kept alive.
+
+    @pytest.mark.parametrize(
+        "error_mode",
+        [
+            None,
+            Manager.BatchElementErrorPolicyTag.kException,
+            Manager.BatchElementErrorPolicyTag.kVariant,
+        ],
+    )
+    def test_when_fail_then_error_emitted(
+        self,
+        manager,
+        a_ref,
+        mock_manager_interface,
+        a_host_session,
+        an_empty_traitsdata,
+        an_entity_trait_set,
+        a_context,
+        a_batch_element_error,
+        invoke_getWithRelationship_error_cb,
+        error_mode,
+    ):
+        page_size = 3
+        method = mock_manager_interface.mock.getWithRelationship
+
+        def call_callbacks(*_args):
+            invoke_getWithRelationship_error_cb(0, a_batch_element_error)
+
+        method.side_effect = call_callbacks
+
+        args = {
+            "entityReference": a_ref,
+            "relationshipTraitsData": an_empty_traitsdata,
+            "pageSize": page_size,
+            "relationsAccess": access.RelationsAccess.kRead,
+            "context": a_context,
+            "resultTraitSet": an_entity_trait_set,
+        }
+
+        if error_mode is not None:
+            args["errorPolicyTag"] = error_mode
+
+        if error_mode is None or error_mode is Manager.BatchElementErrorPolicyTag.kException:
+            with pytest.raises(BatchElementException) as exc_info:
+                manager.getWithRelationship(**args)
+
+            assert exc_info.value.error == a_batch_element_error
+        else:
+            variant_error = manager.getWithRelationship(**args)
+            assert variant_error == a_batch_element_error
+
+
+class Test_Manager_getWithRelationship_batch_convenience:
+
+    @pytest.mark.parametrize(
+        "error_mode",
+        [
+            None,
+            Manager.BatchElementErrorPolicyTag.kException,
+            Manager.BatchElementErrorPolicyTag.kVariant,
+        ],
+    )
+    def test_when_success_then_entityReferencePagers_returned(
+        self,
+        manager,
+        a_ref,
+        mock_manager_interface,
+        a_host_session,
+        an_empty_traitsdata,
+        an_entity_trait_set,
+        a_context,
+        mock_entity_reference_pager_interface,
+        invoke_getWithRelationship_success_cb,
+        error_mode,
+    ):
+
+        two_refs = [a_ref, a_ref]
+        page_size = 3
+        method = mock_manager_interface.mock.getWithRelationship
+
+        def call_callbacks(*_args):
+            invoke_getWithRelationship_success_cb(0, mock_entity_reference_pager_interface)
+            invoke_getWithRelationship_success_cb(1, mock_entity_reference_pager_interface)
```

Comment:
[consider] (Similar cases below) a slightly better test would be if a different value was provided to the callback each time it was called, then we can be confident that the ordering of the output we construct matches the ordering that the manager gave.

Reply:
Nice. Clever use of `hasNext` to disambiguate :+1: